trino-rest-github/src/main/java/pl/net/was/rest/github/Sync.java

```diff
@@ -1293,6 +1298,28 @@ private static void syncCheckRunAnnotations(Options options)
         }
     }
 
+    private static void syncMembers(Options options)
+    {
+        Connection conn = options.conn;
+        String destSchema = options.destSchema;
+        String srcSchema = options.srcSchema;
+        try {
+            conn.createStatement().executeUpdate(
+                    "CREATE TABLE IF NOT EXISTS " + destSchema + ".timestamped_members AS SELECT *, NOW() AS timestamp FROM " + srcSchema + ".members WITH NO DATA");
+            String query = "INSERT INTO " + destSchema + ".timestamped_members " +
+                    "SELECT *, NOW() AS timestamp " +
+                    "FROM unnest(org_members('" + options.org + "'))";
```

Comment:
No SQL injection, please :-) I know the values provided here are explicitly set by a single user, but this code can get flagged in some SAST. Also it would be good to be consistent with other methods in this class.

Reply:
Sync runs pretty often and as I know this won't generate any significant amount of data, I wonder if this can be improved. My idea is to exclude existing members, so you'd have to do a self-join here. By doing a full outer join, we could detect both new and deleted members, and fill out one of two columns, `joined_at` and `removed_at`.

```sql
INSERT INTO dst.timestamped_members
SELECT
  coalesce(src.org, dst.org)
  , coalesce(src.team_slug, dst.team_slug)
  , coalesce(src.login, dst.login)
  , coalesce(src.id, dst.id)
  , coalesce(src.avatar_url, dst.avatar_url)
  , coalesce(src.gravatar_id, dst.gravatar_id)
  , coalesce(src.type, dst.type)
  , coalesce(src.site_admin, dst.site_admin)
  , coalesce(dst.joined_at, current_timestamp) AS joined_at
  , if(src.id IS NULL, current_timestamp) AS removed_at
FROM (SELECT org, team_slug, login, id, avatar_url, gravatar_id, type, site_admin, max(joined_at) AS joined_at, max(removed_at) AS removed_at FROM dst.timestamped_members GROUP BY 1, 2, 3, 4, 5, 6, 7, 8 HAVING max(removed_at) IS NULL) dst
FULL OUTER JOIN unnest(org_members()) src ON (dst.org, coalesce(dst.team_slug, ''), dst.id) = (src.org, coalesce(src.team_slug, ''), src.id)
WHERE dst.id IS NULL OR src.id IS NULL
```

This also supports members that were removed once or more and joined back. The only downside is that we have to mention all columns explicitly so it's verbose :-)

That subquery is what you want to use to get current members. We should provide a view for this in `trino-rest-github/sql` but we shouldn't rely on it to exist in here.

Reply:
Also without this it would be pretty hard to figure out when a user gets removed, as you'd have to group all records to figure out sync dates.